crash-handler-process/platforms/upload-window-win.cpp

```diff
@@ -60,31 +60,40 @@ LRESULT CALLBACK FrameWndProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam)
 	return DefWindowProc(hwnd, msg, wParam, lParam);
 }
 
-void UploadWindow::showButtons(bool ok_enabled, bool yes_enabled, bool cancel_enabled, bool no_enabled)
+void UploadWindow::showButtons(bool ok, bool yes, bool cancel, bool no)
 {
-	ShowWindow(ok_button_hwnd, ok_enabled ? SW_SHOW:SW_HIDE);
-	ShowWindow(yes_button_hwnd, yes_enabled ? SW_SHOW:SW_HIDE);
-	ShowWindow(cancel_button_hwnd, cancel_enabled ? SW_SHOW:SW_HIDE);
-	ShowWindow(no_button_hwnd, no_enabled ? SW_SHOW:SW_HIDE);
+	ShowWindow(ok_button_hwnd, ok ? SW_SHOW:SW_HIDE);
+	ShowWindow(yes_button_hwnd, yes ? SW_SHOW:SW_HIDE);
+	ShowWindow(cancel_button_hwnd, cancel ? SW_SHOW:SW_HIDE);
+	ShowWindow(no_button_hwnd, no ? SW_SHOW:SW_HIDE);
+}
+
+void UploadWindow::enableButtons(bool ok, bool yes, bool cancel, bool no)
+{
+	EnableWindow(ok_button_hwnd, ok );
+	EnableWindow(yes_button_hwnd, yes );
+	EnableWindow(cancel_button_hwnd, cancel );
+	EnableWindow(no_button_hwnd, no );
 }
 
 LRESULT UploadWindow::WndProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam)
 {
 	if (upload_window_hwnd == NULL) {
 		return DefWindowProc(hwnd, msg, wParam, lParam);
 	}
+
 	switch (msg) {
 	case CUSTOM_CLOSE_MSG:
 		log_info << "UploadWindow close message recieved" << std::endl;
 		DestroyWindow(upload_window_hwnd);
 		upload_window_hwnd = NULL;
 		break;
 	case CUSTOM_CAUGHT_CRASH: {
-		SetWindowText(upload_label_hwnd, L"The application just crashed.\n\n"
+		SetWindowText(upload_label_hwnd, L"The application just crashed.\r\n\r\n"
 		"Would you like to share additional information to the developers?");
 		showButtons(false, true, true, false);
+		enableButtons(false, true, true, false);

```

Comment:
can this be more explicit , like passing a struct? a list of 4 bools is not that readable

Reply:
switched our win build to c++20 to have designated initializers.  function call now looks less confusing and not require more lines.